src/test/java/com/example/demo/OrderServiceTests.java

```diff
@@ -1,27 +1,240 @@
 package com.example.demo;
 
 import com.example.demo.matcher.models.OrderAction;
+import com.example.demo.matcher.models.OrderbookItem;
 import com.example.demo.matcher.services.OrderService;
 import com.example.demo.matcher.models.Order;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
 import org.springframework.boot.test.context.SpringBootTest;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.spy;
 
 @SpringBootTest
 public class OrderServiceTests {
     OrderService orderService;
+
     @BeforeEach
     void InitialiseOrderService() {
-        orderService = new OrderService();
+        orderService = spy(new OrderService());
```

Comment:
Good job using `spy` I think you could even use `@Spy` annotation here too. Your preference though as it doesn't really make a difference.

Reply:
I added the annotation next to the declaration of orderService for clarity, but I couldn't figure out how to instantiate it away from the declaration without using spy().